File: docs/source/guides/snippets/grpc/cpp/request.cc
#include "vtsserving/grpc/v1/service.pb.h"

using vtsserving::grpc::v1::VtsService;
using vtsserving::grpc::v1::NDArray;
using vtsserving::grpc::v1::Request;

std::vector<float> data = {3.5, 2.4, 7.8, 5.1};
std::vector<int> shape = {1, 4};

Request request;
request.set_api_name("classify");

NDArray *ndarray = request.mutable_ndarray();
ndarray->mutable_shape()->Assign(shape.begin(), shape.end());
ndarray->mutable_float_values()->Assign(data.begin(), data.end());

File: docs/source/guides/snippets/grpc/swift/Request.swift
import VtsServiceModel

var shape: [Int32] = [1, 4]
var data: [Float] = [3.5, 2.4, 7.8, 5.1]

let ndarray: Vtsml_Grpc_v1_NDArray = .with {
  $0.shape = shape
  $0.floatValues = data
  $0.dtype = Vtsml_Grpc_v1_NDArray.DType.float
}

let request: Vtsml_Grpc_v1_Request = .with {
  $0.apiName = apiName
  $0.ndarray = ndarray
}

